irohad/consensus/yac/storage/impl/yac_proposal_storage.cpp

```diff
@@ -70,8 +70,8 @@ namespace iroha {
           auto iter = findStore(msg.hash.proposal_hash, msg.hash.block_hash);
           auto block_state = iter->insert(msg);
 
-          if (block_state.has_value() and
-              block_state->commit.has_value()) {
+          if (block_state.has_value()
+              and block_state->type() == typeid(CommitMessage)) {
```

Comment:
`typeid` validation can be removed since a single BlockStorage cannot return a reject, because it aggregates votes for a single hash.